src/main/java/com/teamcautionrobotics/robot/Robot.java

```diff
@@ -25,131 +26,154 @@
-   * LiveWindow and SmartDashboard integrated updating.
-   */
-  @Override
-  public void robotPeriodic() {
-    armPower = .5 + .5 * manipulator.getAxis(Axis.LEFT_TRIGGER);
-    hatch.rotate(armPower);
-
-    //if B is pressed, deploy hatch pneumatics
+    VelcroHatch velcroHatch;
+    ExpanderHatch expanderHatch;
+    double armPower;
+    EnhancedJoystick driverLeft, driverRight;
+    Gamepad manipulator;
+    DriveBase driveBase;
+    AimingLights aimingLights;
+    double driveLeftCommand;
+    double driveRightCommand;
+
+    Timer timer;
+
+    boolean deployButtonPressed = false;
+
+    // These are for the Expander Hatch mechanism
+    boolean reacherButtonPressed = false;
+    boolean grabberButtonPressed = false;
+
+    private final boolean usingVelcroHatch = true;
+    private static final String kDefaultAuto = "Default";
+    private static final String kCustomAuto = "My Auto";
+    private String m_autoSelected;
+    private final SendableChooser<String> m_chooser = new SendableChooser<>();
+
+    /**
+     * This function is run when the robot is first started up and should be used
+     * for any initialization code.
+     */
+    @Override
+    public void robotInit() {
+        // pneumatic ports are not finalized
+        velcroHatch = new VelcroHatch(2, 1);
+        expanderHatch = new ExpanderHatch(3, 4);
+        driverLeft = new EnhancedJoystick(0);
+        driverRight = new EnhancedJoystick(1);
+        manipulator = new Gamepad(2);
+        aimingLights = new AimingLights(0, 1);
+        driveBase = new DriveBase(0, 1, 0, 1, 2, 3);
+        CameraServer.getInstance().startAutomaticCapture(0);
+        CameraServer.getInstance().startAutomaticCapture(1);
+
+        timer = new Timer();
+
+        m_chooser.setDefaultOption("Default Auto", kDefaultAuto);
+        m_chooser.addOption("My Auto", kCustomAuto);
+        SmartDashboard.putData("Auto choices", m_chooser);
+    }
+
+    /**
+     * This function is called every robot packet, no matter the mode. Use this for
+     * items like diagnostics that you want ran during disabled, autonomous,
+     * teleoperated and test.
+     *
+     * <p>
+     * This runs after the mode specific periodic functions, but before LiveWindow
+     * and SmartDashboard integrated updating.
+     */
+    @Override
+    public void robotPeriodic() {
+        armPower = .5 + .5 * manipulator.getAxis(Axis.LEFT_TRIGGER);
+        velcroHatch.rotate(armPower);
+
+        // if B is pressed, deploy hatch pneumatics
+
+        driveLeftCommand = driverLeft.getY();
+        driveRightCommand = driverRight.getY();
+
+        aimingLights.set(driverLeft.getRawButton(2));
+
+        if (usingVelcroHatch) {
+            if (manipulator.getButton(Button.B)) {
+                velcroHatch.deploy(manipulator.getButton(Button.B));
+                deployButtonPressed = true;
+            } else {
+                if (deployButtonPressed) {
```

Comment:
Do we want to require releasing the button before the timer starts?

Reply:
We already do.

Reply:
Is this correct behavior?

Reply:
Yes, and it is tested.